crates/ruff/src/rules/flake8_pie/rules.rs

```diff
@@ -392,39 +403,120 @@ pub fn no_unnecessary_dict_kwargs(checker: &mut Checker, expr: &Expr, kwargs: &[
+    for ((attr_name, arg_name), indices) in duplicates {
+        if indices.len() > 1 {
+            let mut diagnostic = Diagnostic::new(
+                SingleStartsEndsWith {
+                    attr: attr_name.to_string(),
+                },
+                Range::from(expr),
+            );
+            if checker.patch(diagnostic.kind.rule()) {
+                let words: Vec<&Expr> = indices
+                    .iter()
+                    .map(|index| &values[*index])
+                    .map(|expr| {
+                        let ExprKind::Call { func: _, args, keywords: _} = &expr.node else {
+                            unreachable!("{}", format!("Indices should only contain `{attr_name}` calls"))
+                        };
+                        args.get(0)
+                            .unwrap_or_else(|| panic!("`{attr_name}` should have one argument"))
+                    })
+                    .collect();
+
+                let call = create_expr(ExprKind::Call {
+                    func: Box::new(create_expr(ExprKind::Attribute {
+                        value: Box::new(create_expr(ExprKind::Name {
+                            id: arg_name.to_string(),
+                            ctx: ExprContext::Load,
+                        })),
+                        attr: attr_name.to_string(),
+                        ctx: ExprContext::Load,
+                    })),
+                    args: vec![create_expr(ExprKind::Tuple {
+                        elts: words
+                            .iter()
+                            .flat_map(|value| {
+                                if let ExprKind::Tuple { elts, .. } = &value.node {
+                                    Left(elts.iter())
+                                } else {
+                                    Right(iter::once(*value))
+                                }
+                            })
+                            .map(Clone::clone)
+                            .collect(),
+                        ctx: ExprContext::Load,
+                    })],
+                    keywords: vec![],
+                });
+
+                // Generate the combined `BoolOp`.
+                let mut first = true;
+                let bool_op = create_expr(ExprKind::BoolOp {
+                    op: Boolop::Or,
+                    values: values
+                        .iter()
+                        .enumerate()
+                        .filter_map(|(index, elt)| {
+                            if indices.contains(&index) {
+                                if std::mem::take(&mut first) {
+                                    Some(call.clone())
+                                } else {
+                                    None
+                                }
+                            } else {
+                                Some(elt.clone())
+                            }
+                        })
+                        .collect(),
+                });
```

Comment:
I tweaked this slightly to ensure that the combined `call` was inserted at the site of the _first_ duplicate index. Previously, it was always inserted at the front of the expression, which led to an unintuitive fix like this:

```diff
diff --git a/setup.py b/setup.py
index 85a2b26357..f20e219ad4 100644
--- a/setup.py
+++ b/setup.py
@@ -236,9 +236,7 @@ def is_macosx_sdk_path(path):
     """
     Returns True if 'path' can be located in a macOS SDK
     """
-    return ( (path.startswith('/usr/') and not path.startswith('/usr/local'))
-                or path.startswith('/System/Library')
-                or path.startswith('/System/iOSSupport') )
+    return ( path.startswith(('/System/Library', '/System/iOSSupport')) or path.startswith('/usr/') and not path.startswith('/usr/local') )
```